resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -258,88 +258,94 @@ public ResteasyProviderFactory(ResteasyProviderFactory parent)
     */
    public ResteasyProviderFactory(ResteasyProviderFactory parent, boolean local)
    {
-      this.parent = parent;
-      providerClasses = new CopyOnWriteArraySet<Class<?>>();
-      providerInstances = new CopyOnWriteArraySet<Object>();
-      properties = new ConcurrentHashMap<String, Object>();
-      properties.putAll(parent.getProperties());
-      enabledFeatures = new CopyOnWriteArraySet<Feature>();
-      reactiveClasses = new ConcurrentHashMap<Class<?>, Class<? extends RxInvokerProvider<?>>>();
-
       if (local)
       {
-         classContracts = new ConcurrentHashMap<Class<?>, Map<Class<?>, Integer>>();
-         if (parent != null)
-         {
-            providerClasses.addAll(parent.providerClasses);
-            providerInstances.addAll(parent.providerInstances);
-            classContracts.putAll(parent.classContracts);
-            properties.putAll(parent.properties);
-            enabledFeatures.addAll(parent.enabledFeatures);
-            reactiveClasses.putAll(parent.reactiveClasses);
-         }
+         // Parent MUST not be referenced after current object is created
+         this.parent = null;
+         initialize(parent);

```

Comment:
@NicoNes, I think that parent should be set to null AFTER initialization. ClientConfiguration calls

>new LocalResteasyProviderFactory(factory);

Reply:
@ronsigal I do not get your point. What will be the difference between 

```
this.parent = null;
initialize(parent);
```
 and that 

```
initialize(parent);
this.parent = null;
```

since  `initialize(parent);` does not use the intance variable `this.parent` but instead the constructor argument `parent`.

Reply:
@NicoNes, D'oh. Right.